packages/grafana-ui/src/components/TimePicker/TimePicker.tsx

```diff
@@ -158,56 +104,52 @@ class UnThemedTimePicker extends PureComponent<Props, State> {
       timeSyncButton,
       isSynced,
       theme,
-      hideText,
+      history,
     } = this.props;
 
+    const { isOpen } = this.state;
     const styles = getStyles(theme);
-    const { isCustomOpen } = this.state;
-    const options = this.mapTimeOptionsToSelectableValues(selectTimeOptions);
-    const currentOption = options.find(item => isTimeOptionEqualToTimeRange(item.value, value));
-
-    const isUTC = timeZone === 'utc';
-
-    const adjustedTime = (time: DateTime) => (isUTC ? time.utc() : time.local()) || null;
-    const adjustedTimeRange = {
-      to: dateMath.isMathString(value.raw.to) ? value.raw.to : adjustedTime(value.to),
-      from: dateMath.isMathString(value.raw.from) ? value.raw.from : adjustedTime(value.from),
-    };
-    const rangeString = rangeUtil.describeTimeRange(adjustedTimeRange);
-
-    const label = !hideText ? (
-      <>
-        {isCustomOpen && <span>Custom time range</span>}
-        {!isCustomOpen && <span>{rangeString}</span>}
-        {isUTC && <span className="time-picker-utc">UTC</span>}
-      </>
-    ) : (
-      ''
-    );
     const hasAbsolute = isDateTime(value.raw.from) || isDateTime(value.raw.to);
 
     return (
-      <div className="time-picker" ref={this.pickerTriggerRef}>
-        <div className="time-picker-buttons">
+      <div className={styles.container}>
+        <div className={styles.buttons}>
           {hasAbsolute && (
             <button className="btn navbar-button navbar-button--tight" onClick={onMoveBackward}>
               <i className="fa fa-chevron-left" />
             </button>
           )}
-          <ButtonSelect
-            className={classNames('time-picker-button-select', {
-              ['explore-active-button-glow']: timeSyncButton && isSynced,
-              [`btn--radius-right-0 ${styles.noRightBorderStyle}`]: timeSyncButton,
-              [styles.timePickerSynced]: timeSyncButton ? isSynced : null,
-            })}
-            value={currentOption}
-            label={label}
-            options={options}
-            maxMenuHeight={600}
-            onChange={this.onSelectChanged}
-            iconClass={classNames('fa fa-clock-o fa-fw', isSynced && timeSyncButton && 'icon-brand-gradient')}
-            tooltipContent={<TimePickerTooltipContent timeRange={value} />}
-          />
+          <div>
+            <Tooltip
```

Comment:
Yhym, this is an interesting choice actually. We usually don't use Tooltip for such scenarios as, well, it's a tooltip. But yeah, we also don't have a guideline about this. I would suggest using the PopoverController here instead, as Tooltip is not a 'semantic' approach.

Reply:
Super feedback! I think I just continued with the same one that we had before. But lets change it.